### PR TITLE
stack.cpp : some changes and improvements

### PR DESCRIPTION
**Pull request by christianbender**
**Changes:**

* Removed shell commands like ```cls``` and ```pause```. Because not platform-independent.  
* Added a destructor for tidy up the heap.  
* Renamed the ```struct node``` in ```struct Node``` according to the [code style conventions](https://github.com/OpenGenus/cosmos/tree/master/guides/coding_style/c%2B%2B)  
* Changed the program logic in the ```main```. Changed the ```while (true)```.  And removed the ```exit(0)```.

### CODE REVIEW

File: code/data_structures/src/stack/stack/stack.cpp [hunk 1]
```diff
@@ -6,19 +6,20 @@
...
-    node *next;
+    Node *next;
 };
 
 template <class T>
 class Stack
 {
 private:
-    node<T> *top;
+    Node<T> *top;
```
Comment (arnavb) [inline code/data_structures/src/stack/stack/stack.cpp]: Should be `Node<T>* top`.

File: code/data_structures/src/stack/stack/stack.cpp [hunk 2]
```diff
@@ -29,9 +30,24 @@ template <class T> Stack<T>::Stack()
...
 
+template <class T> Stack<T>::~Stack()
+{
+    if (top)
+    {
+        Node<T> * ptr = top;
+        Node<T> * tmp = nullptr;
+        while (ptr)
+        {
+            tmp = ptr;
```
Comment (arnavb) [inline code/data_structures/src/stack/stack/stack.cpp]: You can make tmp a local variable.

File: code/data_structures/src/stack/stack/stack.cpp (hunk 1 above)
Comment (christianbender) [reply code/data_structures/src/stack/stack/stack.cpp]: See my commit ```renamed struct node```

**Comment by christianbender** [comment]
@arnavb Done. I made the variable ```tmp``` local.

File: code/data_structures/src/stack/stack/stack.cpp (hunk 1 above)
Comment (arnavb) [reply code/data_structures/src/stack/stack/stack.cpp]: @christianbender I meant that the star should be next to the `<T>`,  not `top`. Same goes for `next` above.

**Comment by christianbender** [comment]
@arnavb Done. I changed the formatting of the pointer type ```Node<T>*```.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus_cosmos

**Pull request merged**